sdk/tests/platform/test_az_platform.c

```diff
@@ -0,0 +1,205 @@
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_timer_start(NULL, 0));
+}
+
+static void test_az_platform_timer_destroy_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_timer_destroy(NULL));
+}
+
+static void test_az_platform_mutex_init_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_mutex_init(NULL));
+}
+
+static void test_az_platform_mutex_acquire_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_mutex_acquire(NULL));
+}
+
+static void test_az_platform_mutex_release_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_mutex_release(NULL));
+}
+
+static void test_az_platform_mutex_destroy_null(void** state)
+{
+  SETUP_PRECONDITION_CHECK_TESTS();
+  (void)state;
+
+  ASSERT_PRECONDITION_CHECKED(az_platform_mutex_destroy(NULL));
+}
+
+#endif // AZ_NO_PRECONDITION_CHECKING
+
+static void test_az_platform_clock_msec_once(void** state)
+{
+  (void)state;
+
+  int64_t test_clock = 0;
+
+  assert_int_equal(az_platform_clock_msec(&test_clock), AZ_OK);
+  assert_true(test_clock >= 0);
+}
+
+static void test_az_platform_sleep_msec_one_second(void** state)
+{
+  (void)state;
+
+  int64_t test_clock_one = 0; // Time before sleep.
+  int64_t test_clock_two = 0; // Time after sleep.
+  int64_t test_expected_time = _az_TIME_MILLISECONDS_PER_SECOND;
+
+  assert_int_equal(az_platform_clock_msec(&test_clock_one), AZ_OK);
+  assert_int_equal(az_platform_sleep_msec(_az_TIME_MILLISECONDS_PER_SECOND), AZ_OK);
+  assert_int_equal(az_platform_clock_msec(&test_clock_two), AZ_OK);
+  
+  assert_true((test_clock_two - test_clock_one) >= test_expected_time);
+}
+
+static void test_az_platform_get_random_once(void** state)
+{
+  (void)state;
+
+  int32_t test_random = 0;
+
+  assert_int_equal(az_platform_get_random(&test_random), AZ_OK);
+  assert_true(test_random > 0);
+}
+
+static void test_az_platform_timer_one_sec(void** state)
+{
+  (void)state;
+
+  az_platform_timer test_timer_handle;
+  int64_t test_clock_one = 0; // Time before timer start.
+  int64_t test_clock_two = 0; // Time after timer callback.
+  int64_t test_expected_time = _az_TIME_MICROSECONDS_PER_MILLISECOND;
+
+  // Create a one second timer.
+  assert_int_equal(az_platform_timer_create(
+        &test_timer_handle,
+        _test_az_platform_timer_callback_set_time,
+        &test_clock_two),
+        AZ_OK);
+  assert_int_equal(az_platform_clock_msec(&test_clock_one), AZ_OK);
+  assert_int_equal(az_platform_timer_start(
+        &test_timer_handle, _az_TIME_MICROSECONDS_PER_MILLISECOND), AZ_OK);
+
+  // Sleep for two seconds until after timer callback is triggered.
+  assert_int_equal(az_platform_sleep_msec(2 * _az_TIME_MICROSECONDS_PER_MILLISECOND), AZ_OK);
```

Comment:
_az_TIME_MILLISECONDS_PER_SECOND?

Reply:
+1 - how is the test passing now?

Reply:
Corrected, tests pass because both _az_TIME_MILLISECONDS_PER_SECOND and _az_TIME_MICROSECONDS_PER_MILLISECOND are 1000.